File: docs-examples/example-java/src/test/java/io/micronaut/nats/docs/consumer/queue/ProductClient.java
package io.micronaut.nats.docs.consumer.queue;

import io.micronaut.context.annotation.Requires;
// tag::imports[]
import io.micronaut.nats.annotation.NatsClient;
import io.micronaut.nats.annotation.Subject;
// end::imports[]

@Requires(property = "spec.name", value = "QueueSpec")
// tag::clazz[]
@NatsClient
public interface ProductClient {

    @Subject(value = "product") // <1>
    void send(byte[] data);
}
// end::clazz[]

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/nats/docs/serdes/ProductClient.kt
package io.micronaut.nats.docs.serdes

import io.micronaut.context.annotation.Requires
import io.micronaut.messaging.annotation.MessageBody
// tag::imports[]
import io.micronaut.nats.annotation.NatsClient
import io.micronaut.nats.annotation.Subject
import io.micronaut.nats.docs.serdes.ProductInfo
import org.reactivestreams.Publisher
// end::imports[]

@Requires(property = "spec.name", value = "ProductInfoSerDesSpec")
// tag::clazz[]
@NatsClient
interface ProductClient {

    @Subject("product")
    fun send(@MessageBody data: ProductInfo)

}
// end::clazz[]
